### PR TITLE
refactor: Integration test syntax improvements

### PR DESCRIPTION
**Pull request by tuliomir**
### Acceptance Criteria
- Improves the test syntax for the #374 tests

### Security Checklist
- [X] Make sure you do not include new dependencies in the project unless strictly necessary and do not include dev-dependencies as production ones. More dependencies increase the possibility of one of them being hijacked and affecting us.

### CODE REVIEW

File: __tests__/integration/hathorwallet_facade.test.js [hunk 1]
```diff
@@ -354,72 +375,66 @@ describe('getFullHistory', () => {
...
-      // Decoded attributes are correct
-      expect(inputObj).toHaveProperty('token', HATHOR_TOKEN_CONFIG.uid);
-      expect(inputObj).toHaveProperty('token_data', TOKEN_DATA.HTR);
-      expect(inputObj).toHaveProperty('script');
-      expect(inputObj).toHaveProperty('value', 10);
-      expect(inputObj).toHaveProperty('decoded');
-      expect(inputObj.decoded).toHaveProperty('type', 'P2PKH');
-      expect(inputObj.decoded).toHaveProperty('address', fundDestinationAddress);
+      expect(moveTx.inputs[inputIndex]).toMatchObject({
+        // Some attributes were correctly translated
```
Comment (pedroferreira1) [inline __tests__/integration/hathorwallet_facade.test.js:397]: Which attributes were not correctly translated?

File: __tests__/integration/hathorwallet_facade.test.js [hunk 2]
```diff
@@ -354,72 +375,66 @@ describe('getFullHistory', () => {
...
-      expect(outputObj).toHaveProperty('selected_as_input', false);
-      expect(outputObj.decoded).toHaveProperty('type', 'P2PKH');
-
-      if (outputObj.value === txValue) {
-        expect(outputObj.decoded).toHaveProperty('address', txDestinationAddress);
-      } else {
-        expect(outputObj.decoded).toHaveProperty('address', txChangeAddress);
-      }
+      expect(outputObj).toMatchObject({
+        // Some attributes were correctly translated
```
Comment (pedroferreira1) [inline __tests__/integration/hathorwallet_facade.test.js:416]: Which attributes were not correctly translated?

File: __tests__/integration/hathorwallet_facade.test.js [hunk 3]
```diff
@@ -1744,10 +1766,13 @@ describe('getTxHistory', () => {
     // HTR transaction incoming
     const tx1 = await GenesisWalletHelper.injectFunds(hWallet.getAddressAtIndex(0), 10);
     txHistory = await hWallet.getTxHistory();
-    expect(txHistory).toHaveLength(1);
-    expect(txHistory[0].txId).toEqual(tx1.hash);
-    expect(txHistory[0].tokenUid).toEqual(HATHOR_TOKEN_CONFIG.uid);
-    expect(txHistory[0].balance).toEqual(10);
+    expect(txHistory).toStrictEqual([
+      expect.objectContaining({
+        txId: expect.any(String),
```
Comment (pedroferreira1) [inline __tests__/integration/hathorwallet_facade.test.js:1771]: Why not `tx1.hash`, as before?

File: __tests__/integration/hathorwallet_facade.test.js [hunk 4]
```diff
@@ -124,54 +124,74 @@ describe('addresses methods', () => {
     for (let i = 0; i < 22; ++i) {
       // Validating generator results
       const genResults = await addressGenerator.next();
-      expect(genResults).toHaveProperty('value');
-      expect(genResults).toHaveProperty('done');
+      expect(genResults).toMatchObject({
```
[the remark this answers is not in the document]
Comment (tuliomir) [reply __tests__/integration/hathorwallet_facade.test.js:127]: I tried adding the `value` field here, but it comes as `undefined` on the last iteration, when `done` is `true`.

To make this check right at this point would get cumbersome, so I decided to validate right below, on lines 135 and 142. This covers all cases for this test.

File: __tests__/integration/hathorwallet_facade.test.js (hunk 3 above)
Comment (tuliomir) [reply __tests__/integration/hathorwallet_facade.test.js]: Fixed it on [dc82fee](https://github.com/HathorNetwork/hathor-wallet-lib/pull/393/commits/dc82fee1edfeb1867d4e95902343c36393456b22), it was supposed to stay as `tx1.hash`

File: __tests__/integration/hathorwallet_facade.test.js (hunk 1 above)
Comment (tuliomir) [reply __tests__/integration/hathorwallet_facade.test.js]: Re-worded this on [dc82fee](https://github.com/HathorNetwork/hathor-wallet-lib/pull/393/commits/dc82fee1edfeb1867d4e95902343c36393456b22)

File: __tests__/integration/hathorwallet_facade.test.js (hunk 2 above)
Comment (tuliomir) [reply __tests__/integration/hathorwallet_facade.test.js]: Re-worded this on [dc82fee](https://github.com/HathorNetwork/hathor-wallet-lib/pull/393/commits/dc82fee1edfeb1867d4e95902343c36393456b22)

**Review by r4mmer — APPROVED** [review]

**Review by pedroferreira1 — APPROVED** [review]

**Pull request merged**